### PR TITLE
Win_services: add exclude filter - 8733 - ignored_names list added

### LINKED ISSUE #8733: Win_services: add exclude filter
## Feature Request
We would like to monitor all services that should run, i.e. are automatic but down. Developers are adding new services quite often, this means maintaining white-list of services is not easy task. Therefore we would like to monitor all (automatic) services and setup alert when some automatic service is down. Create black-list, i.e. list of services we are not interested in is much more easier.

### Proposal:
[[inputs.win_services]]
  ## Names of the services to exclude from monitoring
  exclude = [
     'cloud-init',
     'WbioSrvc'
  ]
  

### Current behavior:

List of services to monitor. Btw there is bug with listed services and related error when service is not installed. When you are monitoring more same HA cluster with (relatively) dynamic distribution of services -> not manageable task.

### Desired behavior:

List of all service not specified in exclude list.

Might be also interesting to specify services with desired startup mode:
 startup_mode = automatic, automatic_delayed

### Use case: 
(copy from above)
Developers are adding new services quite often, this means maintaining white-list of services is not easy task. Therefore we would like to monitor all (automatic) services and setup alert when some automatic service is down. Create black-list, i.e. list of services we are not interested in is much more easier.

### PR DESCRIPTION
**Pull request by let-thomas**
resolves #8733

Not sure how to write/update unit test(s). Could anyone help? I am new Go. And not sure what all is needed on windows to run tests.

### CODE REVIEW

**Comment by let-thomas** [comment]
!signed-cla

**Review by Jagularr — APPROVED** [review]

**Comment by let-thomas** [comment]
!signed-cla

**Comment by let-thomas** [comment]
Can anyone please help me to get over CLA?

**Comment by Jagularr** [comment]
!signed-cla

**Comment by Jagularr** [comment]
Fixed, thanks for signing and the PR

File: plugins/inputs/win_services/win_services.go [hunk 1]
```diff
@@ -94,6 +95,7 @@ type WinServices struct {
 	Log telegraf.Logger
 
 	ServiceNames []string `toml:"service_names"`
+	IgnoredNames []string `toml:"ignored_names"`
```
Comment (sspaink) [inline plugins/inputs/win_services/win_services.go:98]: Would you be ok with renaming this to 'ServiceNamesExcluded' to make it more obvious what the config does?

File: plugins/inputs/win_services/win_services.go [hunk 2]
```diff
@@ -108,7 +110,7 @@ type ServiceInfo struct {
 
 func (m *WinServices) Init() error {
 	var err error
-	m.servicesFilter, err = filter.NewIncludeExcludeFilter(m.ServiceNames, nil)
+	m.servicesFilter, err = filter.NewIncludeExcludeFilter(m.ServiceNames, m.IgnoredNames)
```
Comment (sspaink) [inline plugins/inputs/win_services/win_services.go:113]: To create a unit test for this, you could create a new function similar to 'TestGatherContainsTag' but then test to see if it doesn't contain the tag that you are trying to ignore, the `testutil.Accumulator` comes with a function called `AssertDoesNotContainsTaggedFields` you can use to do this. Hopefully that helps get you started!

**Review by sspaink — CHANGES_REQUESTED** [review]

File: plugins/inputs/win_services/win_services.go (hunk 1 above)
Comment (let-thomas) [reply plugins/inputs/win_services/win_services.go:98]: o.k. no problem

File: plugins/inputs/win_services/win_services.go (hunk 2 above)
Comment (let-thomas) [reply plugins/inputs/win_services/win_services.go]: Test added.

Comment (sspaink) [reply plugins/inputs/win_services/win_services.go]: Great Job!

**Comment by sspaink** [comment]
@let-thomas looks like the tests are failing because you need to run `make fmt`. You might also need to rebase your changes onto the latest master branch because it looks like the CI is running with an older version of Go (should be 1.16).

**Review by sspaink — APPROVED** [review]

**Comment by sspaink** [comment]
Closing because I moved this feature to #10144 to help get it merged, thanks @let-thomas for the work because it just needs some minor changes to get the CI to pass I thought I could help.

**Pull request closed without merging**